tests/test_views.py

```diff
@@ -187,6 +187,49 @@ def test_happy_flow(
         )
         self.assertTrue(User.objects.filter(email="nocollision@example.com").exists())
 
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.get_userinfo")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.store_tokens")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.verify_token")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.get_token")
+    @patch(
+        "mozilla_django_oidc_db.mixins.OpenIDConnectConfig.get_solo",
+        return_value=OpenIDConnectConfig(id=1, enabled=True),
+    )
+    def test_no_userinfo_defined(
+        self,
+        mock_get_solo,
+        mock_get_token,
+        mock_verify_token,
+        mock_store_tokens,
+        mock_get_userinfo,
+    ):
+        """
+        Assert that the login procedure fails gracefully when no user claims are returned
+        """
+        # Testing with empty dict, because mozilla-django-oidc==1.2.x cannot deal with
```

Comment:
@sergei-maertens I noticed that mozilla-django-oidc<2.0 does not deal with a return value for userinfo None` properly here

Reply:
if it's easy to support that, we should, but at some point we'll also have to drop support for < 2.x versions.